hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/storage/HoodieSparkParquetReader.java

```diff
@@ -120,15 +129,56 @@ public ClosableIterator<UnsafeRow> getUnsafeRowIterator(Schema requestedSchema)
   }
 
   public ClosableIterator<UnsafeRow> getUnsafeRowIterator(StructType requestedSchema) throws IOException {
+    return getUnsafeRowIterator(requestedSchema, Collections.emptyList());
+  }
+
+  /**
+   * Read parquet with requested schema and filters.
+   * WARN:
+   * Currently, the filter must only contain field references related to the primary key, as the primary key does not involve schema evolution.
```

Comment:
we can not really filter with non-pk fields right with the event time ordering merging?